lib/iris/common/mixin.py

```diff
@@ -226,12 +224,12 @@ def metadata(self, metadata):
                         if not hasattr(metadata, field)
                     ]
 
-                if missing:
-                    missing = ", ".join(
-                        map(lambda i: "{!r}".format(i), missing)
-                    )
-                    emsg = "Invalid {!r} metadata, require {} to be specified."
-                    raise TypeError(emsg.format(type(arg), missing))
+                    if missing:
```

Comment:
With the `if missing:` check still present, are you worried about the chance of no error being raised if `missing` doesn't get populated? Or is this deliberate?

(I had originally imagined only retaining everything after the `if missing:` check).

Reply:
If I recall correctly, this all depends on what the heck the user has provided (the `rvalue` in the assignment), which could be anything...

I actually refactored this code originally to pull out this specific check, thinking that _clearly_ at this point whatever the user has passed in must be an iterable like a `tuple` or a `list` with the wrong number of elements to match the number of members in the namedtuple metadata instance that is being set (`lvalue` in the assignment).

However, there is a sneaky unit test that creates a custom `class` as the `rvalue` with matching and also missing attributes. For example, see [here](https://github.com/SciTools/iris/blob/v2.4.0/lib/iris/tests/test_cdm.py#L828).

So it is possible that the user has passed a non-iterable, non-mapping as the `rvalue` e.g., a class instance, which may have matching or non-matching attributes.

Hence, the `if missing` is still a valid check.

HTH